gremlin-go/driver/protocol.go

```diff
@@ -45,83 +46,120 @@ type gremlinServerWSProtocol struct {
 	maxContentLength int
 	username         string
 	password         string
+	closed           bool
+	mux              sync.Mutex
 }
 
-func (protocol *protocolBase) connectionMade(transporter transporter) {
-	protocol.transporter = transporter
-}
+func (protocol *gremlinServerWSProtocol) readLoop(resultSets map[string]ResultSet, errorCallback func(), log *logHandler) {
+	for {
+		// Read from transport layer. If the channel is closed, this will error out and exit.
+		msg, err := protocol.transporter.Read()
+		protocol.mux.Lock()
+		if protocol.closed {
+			protocol.mux.Unlock()
```

Comment:
Would it be feasible to add a unit test for this?

Reply:
I don't think so, we don't have any good way of unit testing http comm as of right now... I think it would be difficult to mock in a reasonable amount of time.